common/src/View/CommandProcessor.cpp

```diff
@@ -258,21 +178,21 @@ namespace TrenchBroom {
             m_lastCommandTimestamp = 0;
         }
 
-        CommandProcessor::SubmitAndStoreResult CommandProcessor::submitAndStoreCommand(UndoableCommand::Ptr command, const bool collate) {
+        CommandProcessor::SubmitAndStoreResult CommandProcessor::submitAndStoreCommand(std::unique_ptr<UndoableCommand>&& command, const bool collate) {
```

Comment:
Is this using `std::unique_ptr&&` for the "2. Rvalue Reference Parameter For Conditional Moving" use case described in https://foonathan.net/2018/03/rvalue-references-api-guidelines/ ?

Can it be `std::unique_ptr<UndoableCommand>` instead?

Edit, ok I saw the "this is safe" comment later on, so it looks like the answers are Yes and No. It looks like this is OK but it's confusing/fragile, hopefully the refactor later can get rid of these rvalue ref arguments.

Reply:
fixed